src/ray/raylet/node_manager.cc

```diff
@@ -716,6 +716,14 @@ void NodeManager::ProcessRegisterClientRequestMessage(
   if (message->is_worker()) {
     // Register the new worker.
     worker_pool_.RegisterWorker(std::move(worker));
+    // Push an error message to the user if the worker pool tells us that it is
+    // getting too big.
+    const std::string warning_message = worker_pool_.WarningAboutSize();
+    if (warning_message != "") {
+      RAY_CHECK_OK(gcs_client_->error_table().PushErrorToDriver(
+          JobID::nil(), "worker_pool_large", warning_message, current_time_ms()));
+    }
```

Comment:
It's probably not a big issue. But it might be slightly better to push warnings when we start a new worker (instead of when registering the worker). Because if there're already many worker processes running, the system might be slow, and newly started workers might take long time to register. So warnings would be delayed.

Reply:
Fixed